MetaMask/metamask-mobile
# fix: fix max token amount spend for approvals

## **Description**

During work on our [bridge](https://bridge.aviator.ac/) app, some users noticed that their transactions fail if they try to write a contract and need to approve tokens. We found out that MetaMask-mobile is not correctly setting the MAX token amount because the props provided to the component use `decimalsToShow` = `5` - it was rounding the max amount to 5 decimals, and since it was rounded down, our users weren't able to approve the proper amount of tokens to proceed with transactions.
The solution to that issue is pretty simple: I'm just explicitly setting decimalsToShow to equal the token's decimals. This issue doesn't cause any side effects because token balance is used only for the `CustomSpendCap` component.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Make sure that in your wallet you have a token that uses all of its decimals (or just more than 5)
2. Go to any dApp, and try approving the max amount of the tokens
3. The approved token amount should be equal to your token balance

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
